src/App.tsx

```diff
@@ -1,17 +1,112 @@
-import React from 'react';
-import './App.css';
+import React, { useState } from 'react';
+import './App.scss';
 
+import todos from './api/todos';
 import users from './api/users';
+import { TodoList } from './components/TodoList/TodoList';
+
+const findUserById = (userId: number) => {
+  return users.find(user => user.id === userId) || null;
+};
+
+const prepared = () => {
+  return todos.map(todo => ({
+    ...todo,
+    user: findUserById(todo.userId),
+  }));
+};
 
 const App: React.FC = () => {
+  const [listTodo, setlistTodo] = useState(prepared);
+  const [userId, setUserId] = useState(0);
+  const [usersError, setUsersError] = useState(false);
+  const [titlesError, setTitleError] = useState(false);
+  const [title, setTitle] = useState('');
+
+  const newUserId = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setUsersError(false);
+  };
+
+  const newTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleError(false);
+  };
+
+  const addTodo = () => {
+    if (title.trim() === '') {
+      setTitleError(true);
+    }
+
+    if (userId === 0) {
+      setUsersError(true);
+    }
+
+    if (title.trim() !== '' && userId !== 0) {
+      const newTodo = {
+        userId,
+        id: listTodo.length + 1,
+        title,
+        completed: false,
+        user: findUserById(userId),
+      };
+
+      setlistTodo([...listTodo, newTodo]);
+      setUserId(0);
+      setTitle('');
+    }
+  };
+
   return (
     <div className="App">
-      <h1>Add todo form</h1>
 
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
+      <form onSubmit={(event) => {
+        event.preventDefault();
+        addTodo();
+      }}
```

Comment:
you can call preventDefault inside addTodo method

```suggestion
      <form onSubmit={addTodo}
```